### PR TITLE
Replaces best_hyperparameters

### PR DESCRIPTION
**Pull request by kylebgorman**
The best_hyperparameters.py script is limited because it doesn't understand max/min logic. That is, it cannot report which run achieved the highest validation accuracy, only which had the highest final validation accuracy. However, you can retrieve this information from the W&B website.

Therefore we modify this script to take a run as argument and return its hyperparameters. Sample use:

     ./get_hyperparameters.py cuny-cl/ewt/cvyrtb6d

This returns the hyperparameters for that run, which I previously determined to be the best using W&B's run spreadsheet.

(Note that GitHub choked on what is essentially a file rename with minor changes. `best_hyperparameters.py` and `get_hyperparameters.py` are about 90% the same code; I don't know why it didn't align them like that.)

### CODE REVIEW

File: examples/wandb_sweeps/get_hyperparameters.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        Nested dictionary with proper structure.
+    """
+    nested = {}
+    for key, value in flat_dict.items():
+        if (
+            any(
+                key.startswith(f"{cat}.")
+                for cat in ALLOWED_TOP_LEVEL_CATEGORIES
+            )
+            or key in ALLOWED_TOP_LEVEL_CATEGORIES
```
Comment (DanielYakubov) [inline examples/wandb_sweeps/get_hyperparameters.py:34]: Could you just do

```
any(
                key.startswith(f"{cat}")
                for cat in ALLOWED_TOP_LEVEL_CATEGORIES
            )
```
(without the period)

File: examples/wandb_sweeps/get_hyperparameters.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+
+def main(args: argparse.Namespace) -> None:
+    api = wandb.Api()
+    run = api.run(f"{args.run_path}")
+    logging.info("Run URL: %s", run.url)
+    # Gets all config values from this run.
+    flat_config = {
+        key: value
+        for key, value in sorted(run.config.items())
```
Comment (DanielYakubov) [inline examples/wandb_sweeps/get_hyperparameters.py:47]: why are you sorting before making a dict? Does order matter here? I feel like if so, it would be better to use an explicitly order-preserving type

(I am aware dicts keep insert order in most modern python versions)

File: examples/wandb_sweeps/get_hyperparameters.py (hunk 1 above)
Comment (kylebgorman) [reply examples/wandb_sweeps/get_hyperparameters.py:34]: That would allow false positives like `datathingy` when I want only to match `data.thingy`.
